File: examples/csharp/Scripts/BagScene.cs
using Godot;

namespace FairyGUI.Examples;

public partial class BagScene : DemoSceneBase
{
    private BagWindow _bagWindow;

    protected override void ContinueInit()
    {
        UIPackage.AddPackage("res://Resources/UI/Bag");

        _view = UIPackage.CreateObject("Bag", "Main");
        _groot.AddChild(_view);

        _bagWindow = new BagWindow();

        var bagBtn = _view.GetChild("bagBtn");
        bagBtn?.AddClickListener(Callable.From(() => _bagWindow.Show()));
    }
}

File: examples/ts/Scripts/BagScene.ts
/// <reference path="../fairygui.d.ts" />

import { Callable } from "godot";
import DemoSceneBase from "./DemoSceneBase";
import BagWindow from "./BagWindow";

export default class BagScene extends DemoSceneBase {
    private _bagWindow: BagWindow | null = null;

    protected ContinueInit(): void {
        UIPackage.addPackage("res://Resources/UI/Bag");

        this._view = UIPackage.createObject("Bag", "Main");
        this._groot!.addChild(this._view);

        this._bagWindow = new BagWindow();

        const bagBtn = this._view.getChild("bagBtn");
        if (bagBtn != null) {
            bagBtn.addClickListener(
                Callable.create(() => {
                    this._bagWindow?.show();
                }),
            );
        }
    }
}
